### PR TITLE
Fix dataset.splitter max_history argument

### PR DESCRIPTION
**Pull request by PascalIversen**
*Issue #, if available:*
#1030 

*Description of changes:*

I fixed a bug concerning the `max_history` argument of the dataset.splitter tools and added tests.
(Based on PR #1083)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/gluonts/dataset/split/splitter.py [hunk 1]
```diff
@@ -224,8 +225,10 @@ def split(self, items: List[DataEntry]) -> TrainTestSplit:
             test = self._trim_history(self._test_slice(item))
 
             split._add_train_slice(train)
-
-            assert len(test) - len(train) >= getattr(self, "prediction_length")
+            if (getattr(self, "max_history")) is None:
+                assert len(test) - len(train) >= getattr(
+                    self, "prediction_length"
+                )
```
Comment (lostella) [inline src/gluonts/dataset/split/splitter.py:231]: Why is the guard needed here? Doesn't the condition need to hold in any case?

Comment (PascalIversen) [reply src/gluonts/dataset/split/splitter.py:231]: As I understand it, the reason to use the max_split parameter is to shorten the test set.
For some models it will be enough to use `max_history = context_length+prediction_length`. In that case the test set has length `context_length+prediction_length`, while the train set might be much larger.

Comment (lostella) [reply src/gluonts/dataset/split/splitter.py:231]: > What always holds is something like end_date_train+pd.Timedelta(prediction_length) == end_date_test

Or maybe something like

`end_date_train + prediction_length * end_date_train.freq <= end_date_test`

Comment (lostella) [reply src/gluonts/dataset/split/splitter.py:231]: I think we should keep a meaningful assertion without any condition, because this is essentially checking whether it makes sense to do backtesting on the test data, for the given prediction length, given that you trained on this training data

**Review by lostella — CHANGES_REQUESTED** [review]
Thanks for this @PascalIversen! I think a similar assertion logic should go in the `rolling_split` method, in addition to `split`.

Maybe we can add tests for that method too?

**Review by lostella — APPROVED** [review]
